### PR TITLE
Full Yellow Local - Fix WazuhDB tests self-configuration in 4.2 

### LINKED ISSUE #1535: Full Yellow Local - WazuhDB tests in 4.2
## Description

This issue is part of the https://github.com/wazuh/wazuh-qa/issues/1516. After the research #1493, it has been detected that  `wazuh_db` tests do not properly autoconfigure the testing environment. These tests do not start automatically `wazuh_db` daemon. In an environment where the Wazuh manager has not started for the first time, these tests will fail in some cases.

### Pipeline parameters

|  Jenkins branch  | QA branch   | 
|- |- |
|  4.2 |   1516-4.2.0-full-green |

### Packages details

Type | Format | Architecture | Version | Revision | Tag | File name
-- | -- | -- | -- | -- | -- | --
Server | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-manager-4.2.0-1.1493.x86_64.rpm

In the case presented in the description, wazuh_db tests fail in the following cases:

```

FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Insert commands] - AssertionErro...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Update commands] - AssertionErro...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Select commands] - AssertionErro...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: get-all-agents command] - Assert...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: sync-agent-info-get command] - A...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Reset connection status] - Asser...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: get-agents-by-connection-status command]
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: disconnect-agents command] - Ass...
===================================== 8 failed, 22 passed in 26.11s =====================================
```

The error output obtained is:

```
test_wazuh_db/test_wazuh_db.py ...............FF.FFF..FFF....                                     [100%]
=============================================== FAILURES ================================================
____________________________ test_wazuh_db_messages[global: Insert commands] ____________________________
configure_sockets_environment = None
connect_to_sockets_module = [<wazuh_testing.tools.monitoring.SocketController object at 0x7f7b39d165f8>]
test_case = [{'input': 'global delete-group TestGroup1', 'output': 'ok', 'stage': 'global delete-group previous insert'}, {'input'...lect-groups', 'output': 'ok [{"name":"default"},{"name":"TestGroup1"}]', 'stage': 'global select-groups success'}, ...]
    @pytest.mark.parametrize('test_case',
                             [case['test_case'] for module_data in module_tests for case in module_data[0]],
                             ids=[f"{module_name}: {case['name']}"
                                  for module_data, module_name in module_tests
                                  for case in module_data]
                             )
    def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test_case: list):
        """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
        Parameters
        ----------
        test_case : list
            List of test_case stages (dicts with input, output and stage keys).
        """
        for index, stage in enumerate(test_case):
            if 'ignore' in stage and stage['ignore'] == "yes":
                continue
            expected = stage['output']
            receiver_sockets[0].send(stage['input'], size=True)
            response = receiver_sockets[0].receive(size=True).decode()
            if 'use_regex' in stage and stage['use_regex'] == 'yes':
                match = True if regex_match(expected, response) else False
            else:
                match = (expected == response)
>           assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}' \
                .format(index + 1, stage['stage'], expected, response)
E           AssertionError: Failed test case stage 4: global find-group after insert. Expected: ok [{"id":*}]. Response: ok []
E           assert False
test_wazuh_db/test_wazuh_db.py:105: AssertionError
____________________________ test_wazuh_db_messages[global: Update commands] ____________________________
```

All tests should autoconfigure the testing environment in order to prevent unexpected behaviours.

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|Closes #1535 |

## Description

After research #1493, we could see that WazuhDB tests fail in case Wazuh Manager does not start for the first time. Also, it was detected that in case of the manager has already registered some agents, independently of the state of them, these tests produce false positives. This is error is commented in the [original issue](https://github.com/wazuh/wazuh-qa/issues/1535#issuecomment-883309352). This PR has been created In order to make all wazuh-qa tests independent.

### Packages details

Type | Format | Architecture | Version | Revision | Tag | File name
-- | -- | -- | -- | -- | -- | --
Server | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-manager-4.2.0-1.1493.x86_64.rpm

### Local internal options
No local internal options are required.

### Pytest args
No arguments are required.

## Testing process

In order, to properly test, this issue is necessary to ensure:
- There aren't false positives in a newly provisioned environment, in which the manager has not been started.
- There aren't false positives in an environment in which the manager has already registered agents.

### New provisioned environment

In progress.

### Already registered agents

In progress.

### CODE REVIEW

**Comment by juliamagan** [comment]
I have launched `wazuh_db` tests in branch `1535_full-green-wazuhdb` six times. The first three times, when the manager is started and has already registered agents.

## Tests results

||R1|R2|R3 | Reports|
|:--:|:--:|:--:|:--:|:--:|
|CentOS 8 Manager|  🟢| 🟢 | 🟢 |  [Tests results](https://github.com/wazuh/wazuh-qa/files/6877650/centos_manager_test_wazuh_db.zip) | 

Then, I have launched three times, when the wazuh manager was never started.

||R1|R2|R3 | Reports|
|:--:|:--:|:--:|:--:|:--:|
|CentOS 8 Manager|  🔴 | 🟢 | 🟢 |  [Tests results](https://github.com/wazuh/wazuh-qa/files/6877662/centos_manager_test_wazuh_db.zip) |

File: deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import os
+import subprocess
+import subprocess as sb
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py:3]:
```suggestion
import subprocess as sb
```

It is imported twice, we can delete one import if it is not necessary.

File: deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.wazuh_db import query_wdb
+
+
+def list_agents_ids():
+    wazuhdb_result = query_wdb("global get-all-agents last_id -1")
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py:9]:
```suggestion
    wazuhdb_result = query_wdb('global get-all-agents last_id -1')
```

File: deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            if remove_type == 'manage_agents':
+                subprocess.call([f"{WAZUH_PATH}/bin/manage_agents", "-r", f"{agent_id}"], stdout=open(os.devnull, "w"),
+                                stderr=subprocess.STDOUT)
+            elif remove_type == 'wazuhdb':
+                result = query_wdb(f"global delete-agent {str(agent_id)}")
+
+
+def remove_all_agents(remove_type):
+    agent_ids_list = list_agents_ids()
+    remove_agents(agent_ids_list, remove_type)
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py:30]:
```suggestion
    remove_agents(list_agents_ids(), remove_type)
```
If it is possible, we could call the function `list_agents_ids()` inside `remove_agents` and the code would be simpler.

File: tests/integration/test_wazuh_db/test_wazuh_db.py [hunk 4]
```diff
@@ -56,7 +58,16 @@ def regex_match(regex, string):
     return re.match(regex, string)
```
Comment (juliamagan) [inline tests/integration/test_wazuh_db/test_wazuh_db.py:60]:
```suggestion
# Fixtures
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

**Review by juliamagan — APPROVED** [review]

File: deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py (hunk 1 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py]: Done d2419f50de93aa3c8c7ce0968c6abbe81321f316

File: deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py (hunk 2 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py]: Done d2419f50de93aa3c8c7ce0968c6abbe81321f316

File: deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py (hunk 3 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py]: Done d2419f50de93aa3c8c7ce0968c6abbe81321f316

File: tests/integration/test_wazuh_db/test_wazuh_db.py (hunk 4 above)
Comment (Rebits) [reply tests/integration/test_wazuh_db/test_wazuh_db.py:60]: This comment follows the file standard, however, maybe they are not very useful. It should be necessary to standardize all tests structure and commentaries with the rest of the team in the future.

**Comment by snaow** [comment]
## Tests results

@snaow 

||Status|Reports|
|:--:|:--:|:--:|
|manager/linux| 🟡  | [report_wazuhdb.zip](https://github.com/wazuh/wazuh-qa/files/6998272/report_wazuhdb.zip)|

**Pull request merged**